cas-server-core-events-jpa/src/main/java/org/jasig/cas/config/JpaEventsConfiguration.java

```diff
@@ -0,0 +1,232 @@
+public class JpaEventsConfiguration {
+
+    /**
+     * The Show sql.
+     */
+    @Value("${database.show.sql:true}")
+    private boolean showSql;
+
+    /**
+     * The Generate ddl.
+     */
+    @Value("${database.gen.ddl:true}")
+    private boolean generateDdl;
+
+    /**
+     * The Hibernate dialect.
+     */
+    @Value("${events.jpa.database.dialect:org.hibernate.dialect.HSQLDialect}")
+    private String hibernateDialect;
+
+    /**
+     * The Hibernate hbm 2 ddl auto.
+     */
+    @Value("${events.jpa.database.ddl.auto:create-drop}")
+    private String hibernateHbm2DdlAuto;
+
+    /**
+     * The Hibernate batch size.
+     */
+    @Value("${events.jpa.database.batchSize:1}")
+    private String hibernateBatchSize;
+
+    /**
+     * The Driver class.
+     */
+    @Value("${events.jpa.database.driverClass:org.hsqldb.jdbcDriver}")
+    private String driverClass;
+
+    /**
+     * The Jdbc url.
+     */
+    @Value("${events.jpa.database.url:jdbc:hsqldb:mem:cas-events-registry}")
+    private String jdbcUrl;
+
+    /**
+     * The User.
+     */
+    @Value("${events.jpa.database.user:sa}")
+    private String user;
+
+    /**
+     * The Password.
+     */
+    @Value("${events.jpa.database.password:}")
+    private String password;
+
+    /**
+     * The Initial pool size.
+     */
+    @Value("${events.jpa.database.pool.minSize:6}")
+    private int initialPoolSize;
+
+    /**
+     * The Min pool size.
+     */
+    @Value("${events.jpa.database.pool.minSize:6}")
+    private int minPoolSize;
+
+    /**
+     * The Max pool size.
+     */
+    @Value("${events.jpa.database.pool.maxSize:18}")
+    private int maxPoolSize;
+
+    /**
+     * The Max idle time excess connections.
+     */
+    @Value("${events.jpa.database.pool.maxIdleTime:1000}")
+    private int maxIdleTimeExcessConnections;
+
+    /**
+     * The Checkout timeout.
+     */
+    @Value("${events.jpa.database.pool.maxWait:2000}")
+    private int checkoutTimeout;
+
+    /**
+     * The Acquire increment.
+     */
+    @Value("${events.jpa.database.pool.acquireIncrement:16}")
+    private int acquireIncrement;
+
+    /**
+     * The Acquire retry attempts.
+     */
+    @Value("${events.jpa.database.pool.acquireRetryAttempts:5}")
+    private int acquireRetryAttempts;
+
+    /**
+     * The Acquire retry delay.
+     */
+    @Value("${events.jpa.database.pool.acquireRetryDelay:2000}")
+    private int acquireRetryDelay;
+
+    /**
+     * The Idle connection test period.
+     */
+    @Value("${events.jpa.database.pool.idleConnectionTestPeriod:30}")
+    private int idleConnectionTestPeriod;
+
+    /**
+     * The Preferred test query.
+     */
+    @Value("${events.jpa.database.pool.connectionHealthQuery:select 1}")
+    private String preferredTestQuery;
+
+
+    /**
+     * Jpa event vendor adapter hibernate jpa vendor adapter.
+     *
+     * @return the hibernate jpa vendor adapter
+     */
+    @Bean(name = "jpaEventVendorAdapter")
```

Comment:
Not a huge deal - matter of preference. It's just that duplicating method
name and the same name in annotation value leaves the "bad taste in my
mouth". OK, not a biggie if we stick with the current style.

On Sun, Feb 21, 2016 at 10:39 Misagh Moayyed <notifications@github.com>
wrote:

> In
> cas-server-core-events-jpa/src/main/java/org/jasig/cas/config/JpaEventsConfiguration.java
> <https://github.com/Jasig/cas/pull/1567#discussion_r53568124>:
>
> > +    @Value("${events.jpa.database.pool.idleConnectionTestPeriod:30}")
> > +    private int idleConnectionTestPeriod;
> > +
> > +    /**
> > +     * The Preferred test query.
> > +     */
> > +    @Value("${events.jpa.database.pool.connectionHealthQuery:select 1}")
> > +    private String preferredTestQuery;
> > +
> > +
> > +    /**
> > +     * Jpa event vendor adapter hibernate jpa vendor adapter.
> > +     *
> > +     * @return the hibernate jpa vendor adapter
> > +     */
> > +    @Bean(name = "jpaEventVendorAdapter")
>
> I am trying to remember why I did this. There may have been cases where
> bean names where not valid java method names, and these were names that
> were expected by 3rd party libs. Also, (ah now I remember), there are cases
> where bean names must be defined for aliasing. So I decided to pick one
> style for consistency, but if you want, I'll do scan, and if all checks
> out, will remove names and only leave them where they are needed. (I do
> think the code reads easier this way though, you can immediately tell what
> the bean name is, rather than guessing it must rely on the method name).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jasig/cas/pull/1567/files#r53568124>.
>